jsonrpc/eth_endpoint.go

```diff
@@ -202,8 +203,17 @@ func (e *Eth) GetStorageAt(address types.Address, index types.Hash, number Block
 		}
 		return nil, err
 	}
-
-	return argBytesPtr(result), nil
+	//Parse the RLP value
+	p := &fastrlp.Parser{}
+	v, err := p.Parse(result)
+	if err != nil {
+		return argBytesPtr(types.ZeroHash[:]), nil
+	}
+	data, err := v.Bytes()
+	if err != nil {
+		return argBytesPtr(types.ZeroHash[:]), nil
+	}
```

Comment:
Any reason why we don't return `nil, err` instead of `argBytesPtr(types.ZeroHash[:]), nil` here?
I see that in the previous `if` we return this result, but that is only for the `ErrStateNotFound` error.

Reply:
In case of any internal error related to getStorageAt call, geth return the zero/empty hash, I followed the same. 
Looking forward to your inputs. 😅